dotnet-interactive/Lsp/HoverResponse.cs

```diff
@@ -19,15 +27,30 @@ public HoverResponse(MarkupContent contents, Range range = null)
 
         public static HoverResponse FromHoverEvent(HoverTextProduced responseEvent)
         {
-            var markupContent = responseEvent switch
+            var preferredContent = responseEvent.Content.OrderBy(c => PriorityFromMimeType(c.MimeType)).First();
+            return new HoverResponse(new MarkupContent(MarkupKindFromMimeType(preferredContent.MimeType), preferredContent.Value));
+        }
+
+        private static int PriorityFromMimeType(string mimeType)
```

Comment:
This seems related to content negotiation. Would a more generalized approach make sense in the formatting layer?

Reply:
Not here.  This is the LSP layer where _only_ plaintext or markdown is allowed, not both, and most LSP clients prefer markdown, but given that we plan on removing the LSP shim anyway, this is merely a stopgap until an hour from now when I properly remove LSP.